Рефлексия на предыдущее задание: 

Выполнено практически верно, я опять сделал глупую ошибку и выводил на экран в одном цикле, при обращении к методу foo() в третьем задании. Нужно было вывод делать в отдельном цикле.